internal/services/arckubernetes/arc_kubernetes_provisioned_cluster_resource.go

```diff
@@ -0,0 +1,400 @@
+				Schema: map[string]*pluginsdk.Schema{
+					"admin_group_object_ids": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type:         pluginsdk.TypeString,
+							ValidateFunc: validation.IsUUID,
+						},
+					},
+
+					"azure_rbac_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						Default:  false,
+					},
+
+					"tenant_id": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.IsUUID,
+					},
+				},
+			},
+		},
+
+		"arc_agent_desired_version": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"arc_agent_auto_upgrade_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r ArcKubernetesProvisionedClusterResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"agent_version": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"distribution": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"infrastructure": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"kubernetes_version": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"offering": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"total_core_count": {
+			Type:     pluginsdk.TypeInt,
+			Computed: true,
+		},
+
+		"total_node_count": {
+			Type:     pluginsdk.TypeInt,
+			Computed: true,
+		},
+	}
+}
+
+func (r ArcKubernetesProvisionedClusterResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ArcKubernetes.ArcKubernetesClient
+
+			var model ArcKubernetesProvisionedClusterModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			id := arckubernetes.NewConnectedClusterID(subscriptionId, model.ResourceGroupName, model.Name)
+
+			existing, err := client.ConnectedClusterGet(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			arcAgentAutoUpgrade := arckubernetes.AutoUpgradeOptionsDisabled
+			if model.ArcAgentAutoUpgradeEnabled {
+				arcAgentAutoUpgrade = arckubernetes.AutoUpgradeOptionsEnabled
+			}
+
+			expandedIdentity, err := identity.ExpandSystemAssignedFromModel(model.Identity)
+			if err != nil {
+				return fmt.Errorf("expanding `identity`: %+v", err)
+			}
+
+			payload := &arckubernetes.ConnectedCluster{
+				Identity: *expandedIdentity,
```

Comment:
```suggestion
				Identity: pointer.To(expandedIdentity),
```

Reply:
Fixed, this should be `pointer.From(expandedIdentity)`